server/routes/incidentDetails/incidentDetailsEdit.ts

```diff
@@ -59,37 +109,97 @@ export default class IncidentDetailsEditRoutes {
     const { prisonerNumber, id } = req.params
+    const selectedPerson = JSON.stringify(req.query.selectedPerson)?.replace(/"/g, '')
+    const { originalRadioSelection } = req.session
 
-    const error = validateForm({ incidentDate, locationId })
-    if (error) return this.renderView(req, res, { error, incidentDate, locationId })
+    if (deleteAssociatedPrisoner) {
+      delete req.session.originalRadioSelection
+      delete req.session.redirectUrl
+      // TODO: In here we need to redirect to the delete page, and on the submit button do a PUT request to remove the associated prisoner and incident rolecode, then redirect back to /incident-details/<PRN>/<id>/edit
+      return res.redirect(`/delete-person`)
+    }
+
+    if (search) {
+      const searchValue =
+        search === 'inciteAnotherPrisonerSearchSubmit' ? inciteAnotherPrisonerInput : assistAnotherPrisonerInput
+      const error = validatePrisonerSearch({ searchTerm: searchValue, inputId: currentRadioSelected })
+      if (error)
+        return this.renderView(req, res, {
+          error,
+          incidentDate,
+          locationId,
+          originalRadioSelection: currentRadioSelected,
+          inciteAnotherPrisonerInput,
+          assistAnotherPrisonerInput,
+        })
+      req.session.redirectUrl = `/incident-details/${prisonerNumber}/${id}/edit`
+      req.session.originalRadioSelection = currentRadioSelected
+      return res.redirect(`/select-associated-prisoner?searchTerm=${searchValue}`)
+    }
+
+    const newlySelectedAssociatedPrisonersNumber =
+      isPrisonerIdentifier(selectedPerson) && currentRadioSelected === originalRadioSelection ? selectedPerson : null
+
+    const associatedPrisonersNumber =
+      newlySelectedAssociatedPrisonersNumber || req.session.originalAssociatedPrisonerNumber
+    delete req.session.originalAssociatedPrisonerNumber
+
+    const error = validateForm({
+      incidentDate,
+      locationId,
+      incidentRole: currentRadioSelected,
+      associatedPrisonersNumber,
+    })
+    if (error)
+      return this.renderView(req, res, {
+        error,
+        incidentDate,
+        locationId,
+        originalRadioSelection: currentRadioSelected,
+        associatedPrisonersNumber,
+      })
 
     const IdNumberValue: number = parseInt(id as string, 10)
+    const incidentRoleCode = radioToIncidentRole(currentRadioSelected)
 
     try {
-      const response = await this.placeOnReportService.editDraftIncidentDetails(
+      await this.placeOnReportService.editDraftIncidentDetails(
         IdNumberValue,
         formatDate(incidentDate),
         locationId,
+        associatedPrisonersNumber,
+        incidentRoleCode,
         user
       )
-
-      const nextPageURL = response.draftAdjudication.incidentStatement?.completed
-        ? `/check-your-answers/${prisonerNumber}/${id}`
-        : `/incident-statement/${prisonerNumber}/${id}`
-      return res.redirect(nextPageURL)
+      delete req.session.redirectUrl
```

Comment:
This puzzles me.. Firstly, as above, I can't see why it would be different to the incidentDetails.ts page - so why not reuse?

Secondly, why are 2 items been deleted from the session here and not all of them?